app/services/variables_fetch_service.rb

```diff
@@ -3,32 +3,45 @@
 class VariablesFetchService
   # Load all variables from the OpenFisca server into the database
   #
-  # Updates existing database records but does not remove old records no longer
+  # Updates existing database records and removes old records no longer
   # mentioned by the OpenFisca server
   #
   # @return [Enumerable<Variable>] the loaded +Variable+s, as a streamed
   #   enumerable
   def self.fetch_all
-    variables_list.each do |v|
-      # variables_list returns a small data structure which has the variable
-      # name as a key and other attributes as a hash
-      variable_name = v.first
-      variable_attributes = v.second
+    var_list = variables_list
```

Comment:
Is there a better name for this variable? It seems odd to copy/rename here. If they have good names the reasoning could be clearer.

Reply:
name could definitely be better, reason for re-name is because variable list is a function and we want to set the result of it to static in order to not call it twice

Reply:
as been refactored